src/dbnode/namespace/dynamic.go

```diff
@@ -107,22 +107,49 @@ func newDynamicRegistry(opts DynamicOptions) (Registry, error) {
 	}
 
 	logger := opts.InstrumentOptions().Logger()
-	logger.Info("waiting for dynamic namespace registry initialization, " +
-		"if this takes a long time, make sure that a namespace is configured")
-	<-watch.C()
+	var (
+		watchable = xwatch.NewWatchable()
+		initValue kv.Value
+		dt        *dynamicRegistry
+	)
+	if opts.AllowEmptyInitialNamespaceRegistry() {
+		initValue, err = kvStore.Get(opts.NamespaceRegistryKey())
+		if err == kv.ErrNotFound {
+			logger.Info("no initial namespaces found. namespaces will be added via watch updates")
+			dt = &dynamicRegistry{
+				opts:      opts,
+				logger:    logger,
+				metrics:   newDynamicRegistryMetrics(opts),
+				watchable: watchable,
+				kvWatch:   watch,
+			}
+
+			startRegistryWatch(dt)
+
+			return dt, nil
+		} else if err != nil {
+			return nil, err
+		}
+	} else {
+		logger.Info("waiting for dynamic namespace registry initialization, " +
+			"if this takes a long time, make sure that a namespace is configured")
+
+		<-watch.C()
+
+		initValue = watch.Get()
```

Comment:
Nah, this `dynamicInitializer` is only really used _after_ you've decided to use etcd based namespace resolution (which for dbnodes has always been the case, and, now, for coordinators, is something we can opt into). Line 137 is probably the most succinct way to express what the flag controls in that it will block indefinitely waiting for namespaces. If you want that behavior, which dbnodes that are starting up do, then you shouldn't set the flag. Whereas coordinators do not since you want to be able to create namespaces from an empty state (e.g. when you first create your m3 cluster).